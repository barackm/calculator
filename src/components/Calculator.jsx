import React, { useState } from 'react';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState('0');
  const [operation, setOperation] = useState(null);

  const handleClick = (buttonName) => {
    const result = calculate({ total, next, operation }, buttonName);
    setTotal(result.total);
    setNext(result.next);
    setOperation(result.operation);
  };

  return (
    <div className="calculater-main-container">
      <Display result={next} />
      <ButtonPanel clickHandler={handleClick} />
    </div>
  );
};

export default Calculator;
