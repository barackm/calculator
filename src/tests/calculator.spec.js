import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculator Component', () => {
  beforeEach(() => {
    render(<Calculator />);
  });
  it('Should match spanshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should give the sum of two clicked numbers with the + button', async () => {
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByTitle('result')).toHaveTextContent('10');
  });
  it('should give the product of two clicked numbers with the x button', async () => {
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByTitle('result')).toHaveTextContent('25');
  });
});
