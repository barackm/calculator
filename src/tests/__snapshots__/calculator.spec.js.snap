// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component Should match spanshot 1`] = `
<div
  className="calculator-wrapper d-flex justify-content-center align-items-center px-5"
>
  <div
    className="overlay-calculator"
  />
  <div
    className="calculator-left-area"
  >
    <h3>
      Make all your Calculations Here
    </h3>
    <hr
      className="diviser"
    />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Necessitatibus magni velit tempora quam ut tempore totam sapiente quod voluptatem soluta. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quasi, magni minima omnis adipisci excepturi amet quaerat commodi tempora hic enim voluptas quae quisquam consequuntur? Harum ducimus nulla explicabo fugiat aspernatur?
    </p>
  </div>
  <div
    className="calculater-main-container"
  >
    <div
      className="display-container"
    >
      <h3
        title="result"
      >
        0
      </h3>
    </div>
    <div
      className="btn-group"
    >
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="btn-group"
    >
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="btn-group"
    >
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="btn-group"
    >
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="btn-group"
    >
      <button
        className="zero"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
